src/enzyme_ad/jax/Implementations/HLODerivatives.td

```diff
@@ -541,15 +543,77 @@ def GradEinsumConfigRhs : GlobalExpr</*needsprimal*/0, /*needsshadow*/0, [{
 }]>;
 
 // Derivative rules
+def : HLODerivative<"AcosOp", (Op $x), [
+  (Neg (CheckedDiv (DiffeRet), (Sqrt (Sub (HLOConstantFP<"1">), (Pow $x, (HLOConstantFP<"2">))))))
+]>;
+
+def : HLODerivative<"AcoshOp", (Op $x), [
+  (CheckedDiv (DiffeRet), (CheckedMul (Sqrt (Sub (Pow $x, (HLOConstantFP<"2">)), (HLOConstantFP<"1">))), (Sqrt (Add (Pow $x, (HLOConstantFP<"2">)), (HLOConstantFP<"1">)))))
+]>;
+
 def : HLODerivative<"AddOp", (Op $x, $y),
                     [
                       (DiffeRet),
                       (DiffeRet),
                     ]
                   >;
 
+def : HLOInactiveOp<"AfterAllOp">;
```

Comment:
This should be handled later since properly handling parallelism means reversing the parallel dependencies